sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_providers/local_provider.py

```diff
@@ -63,6 +64,17 @@ def _raise_if_unsupported(self, operation, algorithm):
     def encrypt(self, algorithm, plaintext, iv=None):
         # type: (EncryptionAlgorithm, bytes, Optional[bytes]) -> EncryptResult
         self._raise_if_unsupported(KeyOperation.encrypt, algorithm)
+
+        # If an IV isn't provided with AES-CBCPAD encryption, try to create one
+        if iv is None and algorithm.value.endswith("CBCPAD"):
```

Comment:
Could also end with "CBC".

Reply:
We only support CBCPAD locally, unfortunately -- Python's cryptography library doesn't have a supported padding mode for CBC encryption. It would be theoretically straightforward to implement, but doing our own padding implementation feels a bit risky since I'm not an expert